addOns/network/src/main/java/org/zaproxy/addon/network/internal/ui/ratelimit/NewRateLimitRuleContextMenu.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2022 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.addon.network.internal.ui.ratelimit;
+
+import org.parosproxy.paros.control.Control;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.addon.network.ExtensionNetwork;
+import org.zaproxy.zap.view.messagecontainer.http.HttpMessageContainer;
+import org.zaproxy.zap.view.popup.PopupMenuItemHttpMessageContainer;
+
+/**
+ * Base context menu item to add rate limit rule.
+ *
+ * @see HttpMessageContainer
+ */
+@SuppressWarnings("serial")
+public abstract class NewRateLimitRuleContextMenu extends PopupMenuItemHttpMessageContainer {
+    @SuppressWarnings("unused")
+    protected final ExtensionNetwork extension;
+
+    public NewRateLimitRuleContextMenu(String label, ExtensionNetwork ext) {
```

Comment:
Should pass `RateLimitOptions` which is what's required, it will also make the tests a lot simpler.